### PR TITLE
#3909 Add GliderLink support

### PR DESCRIPTION
**Pull request by wargoth**

### CODE REVIEW

File: src/GliderLink/GliderLinkId.hpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * The identification number of a GLIDER_LINK traffic.
+ */
+class GliderLinkId {
+  uint32_t value;
+
+public:
+  constexpr
+  GliderLinkId(uint32_t _value):value(_value) {}
```
Comment (MaxKellermann) [inline src/GliderLink/GliderLinkId.hpp]: Make this constructor `explicit`!

File: src/GliderLink/GliderLinkId.hpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  GliderLinkId() = default;
+
+  bool operator==(GliderLinkId other) const {
```
Comment (MaxKellermann) [inline src/GliderLink/GliderLinkId.hpp]: Add `constexpr`

File: src/MapWindow/MapWindowTraffic.cpp [hunk 3]
```diff
@@ -115,6 +115,95 @@ MapWindow::DrawFLARMTraffic(Canvas &canvas,
...
+{
+  // Return if FLARM icons on moving map are disabled
+  if (!GetMapSettings().show_flarm_on_map)
+    return;
+
+  const GliderLinkTrafficList &traffic = Basic().glink_data.traffic;
+  if (traffic.IsEmpty())
+    return;
+
+  const MoreData basic = Basic();
```
Comment (MaxKellermann) [inline src/MapWindow/MapWindowTraffic.cpp]: Why copy the whole `MoreData` struct?

File: src/MapWindow/MapWindowTraffic.cpp [hunk 4]
```diff
@@ -115,6 +115,95 @@ MapWindow::DrawFLARMTraffic(Canvas &canvas,
...
+
+  const WindowProjection &projection = render_projection;
+
+  canvas.Select(*traffic_look.font);
+
+  // Circle through the GliderLink targets
+  for (auto it = traffic.list.begin(), end = traffic.list.end();
```
Comment (MaxKellermann) [inline src/MapWindow/MapWindowTraffic.cpp]: Use range-based `for`

File: src/GliderLink/Traffic.hpp [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  RoughAngle track;
+
+  /** Speed of the GliderLink target */
+  RoughSpeed speed;
+
+  /** Altitude of the GliderLink target */
+  RoughAltitude altitude;
+
+  /** Climb rate of the GliderLink target */
+  fixed climb_rate;
```
[the remark this answers is not in the document]
Comment (wargoth) [reply src/GliderLink/Traffic.hpp]: Copied this from FLARM implementation. What should be used instead?

[the remark this answers is not in the document]
Comment (MaxKellermann) [reply src/GliderLink/Traffic.hpp]: You copied from branch v6.8.x, but this doesn't exist in master, and won't compile with master.

**Comment by MaxKellermann** [comment]
Thanks for the fixes, but there are still conflicts (GitHub shows message "This branch has conflicts that must be resolved"). Please rebase your branch on our master branch and fix all the conflicts.

**Comment by MaxKellermann** [comment]
And there are still conflicts. Do you understand what this problem is about?

**Comment by wargoth** [comment]
Hi Max,

Absolutely! I'm still working on refactoring and using 6.8.x to test since
I haven't managed to build master yet. Afterwards I'll rebase the changes.
It might be easier to abandon this PR, I can create a new one later.

Thank you,
-Val

**Comment by MaxKellermann** [comment]
I think you should rebase before you do anything else, because the more code you add, the more work will the rebase be. Do it now and save time.
And you don't need to abandon this PR. Just force-push your new commit to the PR branch.

**Comment by MaxKellermann** [comment]
Closing because the stable branch shouldn't get new features, only bug fixes. See #143

**Pull request closed without merging**